### DISCUSSION TITLE
Scripting Execution

### DISCUSSION THREAD

**Discussion by njrusty** [JVital2013/vitality-goes#84] [General]
Apologizes all but I have been scanning all the documentation and help files and I cannot find how you actually 'execute' a script. I've found the scripts folder and am interested in the scripts to cleanup files. I've modified my scriptconfig.ini but I am now sure how it gets executed.  I use the following to run my system:

satdump live goes_hrit GOES_16 --source rtlsdr --samplerate 2.4e6 --frequency 1694.1e6 --gain 49 --http_server 192.168.1.8:8080 --fill_missing

Would appreciate any help.

**Comment by stonegrift** [comment]
The scripts are "executed by taking them out and shooting them.  No really, they are what is referred to as a bash script.  and they are executed by simply typing (no quotes) "bash scriptname.sh"  where .sh is the extension.  So for CreateVidieos-ABI-goestools.sh  you would type "bash CreateVideos-ABI-goestools.sh" and hit enter in a terminal window. You may or may not need to add the path to the script.  Being an old windows user, I'm always moving to the directory the file is located in before I run it.  Some of these scripts will complain and give you instructions if they are missing programs, like zip, imagemagick ffmpeg and so on.  The complaint generally will give you the instruction to install missing software.

**Comment by njrusty** [comment]
Appreciate the help. I've been playing with this for an hour and have given up. No matter what script I executed, I still get a No file or directory message. But I've followed all the instructions (from above) and in documentation but obviously I am missing something.  I am hoping someone will post their config file.

I've been entering in a command line:
bash Delete-Old-SatDump.sh
bash: Delete-Old-SatDump.sh: No such file or directory

**Comment by stonegrift** [comment]
Sitting here entering all this information and I forgot the most important.  Are you trying to run these scripts on a windows machine or Linux?

If it's linux, you can keep reading.  IF it's windows, you're messing with the wrong scripts.

Now I should point out, even in my own case, if there are none of the files it's looking for. the scripts will throw errors.  The CreateVideos-EMWIN.sh file ALWAYS throws an error when it attempts to do the Japanese satellite It complains the file is missing.  But it keeps going.  

What is your scriptconfig.ini ? Actually, never mind. Let's start from scratch.

Rename the existing scriptconfig.ini.  Now let's make a new one.  I want you to make one with the bare minimum you need to test with.
Assuming the videoDir is correct,. you'll just hit the other key elements.
-------------------------------
videoDir="/var/www/html/videos"

abiSrcDir="/path/to/goestoolsrepo"
abiImgSource=("$abiSrcDir/goes18/fd/fc")
abiVidName=("GOES18FalseColor"")
---------------------------------

So you will need the quote marks. for the variables.
so abiSrcDir="/home/YOURUSERNAME/ followed by the name of your goes image storage

IF you followed USradioguy.com's guide for receiving images from goestools  He set the output directory to "/home/pi/goes"  where you replace pi with your username  SO abiSrcDir="/home/pi/goes"

abiImgSource is just pointing at the full color full disk image for the satellite you're looking at. and setting the name as GOES18FalseColor  or GOES16FalseColor Depending on which one you're setting up for

Now save it.
I personally changed the name of the two create video sh files to remove which one it was made for.  So I have them named CreateVideos-ABI.sh and CreateVideos-EMWIN.sh  We are just worried about the ABI file.

Another thing to remember is that the script for the create videos are very different between SatDump and Goestools.  the two programs DO NOT save with the same folder structure.  

Now before you run this test of the script config, make sure you have images saved.   Since you may not have changed file names for the CreateVideos-ABI-*.sh type. just do the bash with the correct version.  The short scriptconfig.ini will keep it quick

Once it runs correctly once, you can then repeat the process with the second.   ALSO

**Comment by njrusty** [comment]
Apologizes, was away for some time. 

I am running SatDump on an RP4 so I am running Linux. Trying to convert your instructions above but for SatDump. Not sure what goestoolsrepo is???  Also, the original scriptconfig.ini wants a folder location for ffmpeg.exe which I cannot find anywhere on the RP4 system.

Also, as an example; my image files are actually located at:  /home/pi/GOES_16/IMAGES/GOES-16/Full Disk/   (FOR FULL DISK IMAGES) 
and emwinSrcDir="/home/pi/GOES_16/EMWIN"  (FOR THE EMWIN FILES)

so I assume the abiSrcDir location will be:  abiSrcDir="/home/pi/GOES_16"  ?  Unfortunately I have tried many combos with the command 

bash Cleanup-EmwinText.sh

And I still keep getting an error message for no such file or directory.

**Reply by stonegrift** [reply]
No problem the goestoolsrepo is the creator's shorthand for where the satellite data (images and text) are stored. The ffmpeg stuff is where it wants the path to the program itself. in my case it's /usr/bin/ffmpeg.  abiSrcDir is the same thing as the goestools repo. the root spot you have it stored. In my case abiSrcDir="/home/pi/goes"  abiSrcDir is a short term variable to replace having to write real long entries in the abiImgSource array list.  It's easier to write $abiSrcDir/.. instead of /home/pi/goes/...   As you can guess the 3 lines with abi in front of them deal with making the videos from the ABI.ini vitality page. and the ones prefaced with emwin does the same thing for the emwin.ini page.  The difference is that for emwin it only has to point at ONE directory rather than the root since the ABI page deals with so many different channels of images.  Satdump is different and has different scripting because Satdump stored the data a bit differently.

Remember Linux pays attention to letter case.  So CleanUp-EmwinText.sh is different that cleanup-EmwinText.sh

**Comment by njrusty** [comment]
Appreciate your help and it appears I have most of it working. However; I had to copy my scriptconfig.ini file and put it in the folder /vitality-goes/scripts-linux in order for the bash command to work.

**Comment by stonegrift** [comment]
I should have pointed that out. Personally, I put all the scripts and config files off the Apache web space folder. Ie /var/www/html  I added a sub folder called config so it was /var/www/html/config.  Under that the Sanchez script added a "resources" directory with some of the pictures needed for the underlays.  The only other way I could think of to get at them without putting them in the same folder is adding the config files to the path, which I really don't know how to do just yet.  I've just recently been delving into the bash scripting trying to see if there is an easier way to organize the source images and the mp4 file names.

**Comment by stonegrift** [comment]
Well, It's not much since I only just started digging and it's a sideline.  Here are the most of the variables in Config.ini and Scriptconfig.ini

videoDir :  Directory where the videos made by the scripts is to be stored.
abiSrcDir: ABI image source directory. The root of your goes satellite data.
abiImgSource: Using abiSrcDir as a lead in for the path to where images are stored you want to animate. Such as $abiSrcDir/goes18/fd/fc for goes 18 full disk full color images
abiVidNames: The name you want associated with each image set you want to make a video of.

emwinSrcDir: Path to the emwin data folder (to the actual emwin folder)
emwinCodeName : Similar to the abi entries another source and destination for EMWIN videos
emwinVideoName: Similar to the abi entries another source and destination for EMWIN videos

These are in config.ini
graphiteAPI: URL for tracking your Pi with graphite
fastEmwin: flag variable for using Vitality Goes on a Windows machine
spaceWeatherAlerts: flag to track solar weather
showSysInfo: flag to show server information
debug: Flag for advanced program debugging.

Path
GOES18 = /home/pi/goes        This varies depending on which satellite you're looking at or using Vitality goes to view. You can even have multiple Paths. Each to a different folder 

That's it for now.

**Comment by radarcharlie** [comment]
Hello, for the record, I want to thank the developers for this wonderful application.  Like the OP, I'm having issues executing the create video script for Windows; or any of the Windows scripts for that matter.  I have combed the web and found little to  nothing on Windows script execution.  I chose a Vitality Goes Windows setup because it was the quickest was to experience Vitality Goes without building a Linux box.   Everything is running perfectly except the Windows script execution.  To this point after many  iterations of trying to determine where to place the scripts, attempts at executing the scripts using Powershell have only led to frustration and errors.  If anyone can point me in the right direction it would be greatly appreciated.  The config files are pretty much straight forward.  So what I have configured in scriptconfig.ini is as follows:

videoDir="C:\xampp\htdocs\videos"
ffmpegBinary="C:\ffmpeg\bin\ffmpeg.exe"

abiSrcDir="F:\NOAA APT\2024-10-30_01-20_goes_hrit_1.6941 GHz"

[config.ini]

[general]
;siteTitle = "My GOES-16 Data"
;siteTheme = "light"
;satdumpAPI = http://127.0.0.1:8080/api
adminPath = "F:\NOAA APT\2024-10-30_01-20_goes_hrit_1.6941 GHz\Admin Messages"
emwinPath = "F:\NOAA APT\2024-10-30_01-20_goes_hrit_1.6941 GHz\EMWIN"
fastEmwin = true
spaceWeatherAlerts = false
showSysInfo = true
debug = false

[paths]
GOES16 = "F:\NOAA APT\2024-10-30_01-20_goes_hrit_1.6941 GHz"

I get the timelapse window, a couple of spins like something is happening and nothing after that.  I looks like the video creation script is not executing on the surface.  To  be honest, I have no idea when to  place the scripts.  Again I would greatly appreciate some guidance on how to execute the windows scripts.  Thanks in advance.

**Reply by radarcharlie** [reply]
NO one can offer any help?
